tracer/src/Datadog.Trace/Agent/AgentWriter.cs

```diff
@@ -384,18 +384,28 @@ SpanBuffer SwapBuffers()
                 return null;
             }
 
-            trace = _statsAggregator?.ProcessTrace(trace) ?? trace;
-            bool forceKeep = _statsAggregator?.AddRange(trace) ?? false;
-
-            // If stats computation determined that we can drop the P0 Trace,
-            // skip all other processing
-            if (!shouldSerializeSpans && CanComputeStats && !forceKeep)
+            // Eagerly return if the trace is empty
+            if (trace.Count == 0)
             {
-                Interlocked.Increment(ref _droppedP0Traces);
-                Interlocked.Add(ref _droppedP0Spans, trace.Count);
                 return;
             }
 
+            trace = _statsAggregator?.ProcessTrace(trace) ?? trace;
+            if (CanComputeStats)
```

Comment:
if `CanComputeStats` is `false`, do we need to run `_statsAggregator?.ProcessTrace()`? Can we move it inside the `if`?

Reply:
I see you do this in https://github.com/DataDog/dd-trace-dotnet/pull/3152, so can ignore this 🙂